PenLineModel.cpp

```diff
@@ -140,10 +179,31 @@ void PenLineModel::addPoint(int lineIndex, PenPoint point)
 
 }
 
+void PenLineModel::undo() {
+    m_undoStack.undo();
+    emit undoStackChanged();
+}
+
+void PenLineModel::redo() {
+    m_undoStack.redo();
+    emit undoStackChanged();
+}
+
 void PenLineModel::clear() {
+    if (m_activeUndoCommand != nullptr) {
+        delete m_activeUndoCommand;
```

Comment:
This will likely cause a heap-use after free runtime error because the qundostack owns commands. From the qt docs: Once a command is pushed, the stack takes ownership of it.

https://doc.qt.io/qt-6/qundostack.html#push

Reply:
Whoops! Stupid mistake, good catch.